### ISSUE TITLE
Support client certificates

### ISSUE THREAD

**Issue by DocMoebiuz** [rpdswtk/vsmqtt#163]
At the moment I can see that there is support for a CA certificate to trust the server. However, looking at the source code the options for client certificate authentication is not provided

This is from the TLS library (https://nodejs.org/api/tls.html#tls_tls_connect_options_callback)
```
const options = {
  // Necessary only if the server requires client certificate authentication.
  key: fs.readFileSync('client-key.pem'),
  cert: fs.readFileSync('client-cert.pem'),

  // Necessary only if the server uses a self-signed certificate.
  ca: [ fs.readFileSync('server-cert.pem') ],

  // Necessary only if the server's cert isn't for "localhost".
  checkServerIdentity: () => { return null; },
};
```

So we should support the options for key and cert, it is basically the same approach as for the server certificate handling right now
https://github.com/rpdswtk/vsmqtt/blob/f707d73e77c1662dbc786527aab03c87d423582b/src/MqttClientFactory.ts#L19

**Comment by rpdswtk** [comment]
Thanks for your contribution I will release it this weekend.

**Comment by DocMoebiuz** [comment]
cool!

**Comment by DocMoebiuz** [comment]
#146 is another option that could be exposed, i will try to create another PR

**Comment by rpdswtk** [comment]
This is now released with 1.4.3

**Comment by DocMoebiuz** [comment]
Great! Thanks.
